docs/api/en/lights/PointLight.html

```diff
@@ -67,8 +66,18 @@ <h3>[property:Float decay]</h3>
 
 		<h3>[property:Float distance]</h3>
 		<p>
-			If non-zero, light will attenuate linearly from maximum intensity at the light's
-			position down to zero at this distance from the light. Default is *0.0*.
+			<em>Default mode</em> — When distance is zero, light does not attenuate. When distance is
+			non-zero, light will attenuate linearly from maximum intensity at the light's position down to
+			zero at this distance from the light.
+		</p>
+		<p>
+			<em>[page:WebGLRenderer.physicallyCorrectLights Physically correct] mode</em> — When distance
+			is zero, light will attenuate according to inverse-square law to infinite distance. When
+			distance is non-zero, a more sudden cutoff is applied at this distance from the light, in
```

Comment:
>a more sudden cutoff is applied

@donmccurdy Are you sure that is true? I thought we specifically ensured that it decayed to zero smoothly, regardless.

Reply:
Yes to both — the cutoff is smooth and sudden near the `distance` threshold. From the Frostbite paper:

>  At a certain limit, the illuminance should smoothly reach zero. One approach to solve this is to window the falloff in such a way that most of the function is unaffected.

I believe that's what we've done:

```glsl
if( cutoffDistance > 0.0 ) {

	distanceFalloff *= pow2( saturate( 1.0 - pow4( lightDistance / cutoffDistance ) ) );

}
```

Reply:
I recall we specifically did not want a sudden cutoff. I think it attenuates to zero smoothly. It's a hack.

I would say it is

    inverse-squared falloff, attenuating to zero intensity at the cutoff distance.

That is a reasonably-accurate description. Maybe you can think of a more-accurate way to say it...

Reply:
The essential thing is that the cutoff's smoothness is not distributed evenly across the light's range (as it is in non-physical mode). Note in [this demo](https://pointlighttest-flixoveswh.now.sh/examples/lighttest) that once the cutoff is >5, it's essentially invisible.

The glTF spec's wording matches the Frostbite and Unity approach as well, I think:

> Within the range of the light, attenuation should follow the inverse square law as closely as possible, although some non-quadratic falloff near the edge of the range may be used to avoid a hard cutoff.

So the goal is for the cutoff to be as smooth as possible, given constraints of (1) not affecting too much of the light's range, and (2) intensity=0 beyond the cutoff.

😅 Trying again on wording, how about this:

> When distance is non-zero, light will attenuate according to inverse-square law until near the distance cutoff, where it will then attenuate quickly and smoothly to 0. While cutoffs are inherently not physically correct, they are useful for limiting the number of lights affecting any given object.

Reply:
@donmccurdy Yes, that's better.

> they are useful for limiting the number of lights affecting any given object.

I wonder if that really is only a benefit to a deferred renderer... 

In our forward renderer, my guess is cutoffs are useful primarily with lights that cast shadows, where the shadow camera far plane can cause shadows to be clipped if they have not attenuated completely.

Reply:
True. If https://github.com/mrdoob/three.js/issues/9325 were resolved in the forward renderer it would be another story, but I haven't looked at this closely.

Reply:
Me either. For now, I'd eliminate the "useful for limiting the number of lights" part...

Thanks for this. :-)